holmes/core/llm.py

```diff
@@ -186,6 +191,7 @@ def completion(
             temperature=temperature,
             response_format=response_format,
             drop_params=drop_params,
+            thinking=thinking,
```

Comment:
I would suggest renaming `tool_args` to `extra_args` and then:

```
        if THINKING:  # if model requires 'thinking', load it from env vars
            extra_args["thinking"] = json.loads(THINKING)
```

This is because litellm sometimes treat present but `None` arguments differently than when arguments are not present. I doubt it'll be an issue in this case though but still suggesting more defensive code.

Reply:
Looks like this behaves like the default, so I think it's ok in this case